lisp/system.lisp

```diff
@@ -17,13 +17,20 @@ Mahogany is running under")
 		     (uiop:run-program (list "which" name) :output stream)))
     (UIOP/RUN-PROGRAM:SUBPROCESS-ERROR nil)))
 
+(defun open-program (candidates)
+  (loop :for name :in candidates
+		:do (alex:if-let ((program (find-program name)))
+              (progn
+                (uiop:launch-program program)
+                (return t))
+              (log-string :warn "Could not find program ~S in candidates ~S." name candidates)))
+  (values nil))
+
+(defvar *default-terminals*
```

Comment:
I missed this on the first round, but if this is a variable that a user will eventually be able to modify, there's a macro for that called `defconfig`. It would look something like:

``` lisp
(config-system:defconfig *default-terminals*
  (list ...)
  list
  "A list of default terminal programs to use")
```
It wraps `defvar` with some machinery to record the symbol name, the initial value, and the type of the variable.